scripts/dev/k8sutil.py

```diff
@@ -0,0 +1,63 @@
+import time
+
+from kubernetes.client.rest import ApiException
+
+# time to sleep between retries
+SLEEP_TIME = 2
+# no timeout (loop forever)
+INFINITY = -1
+
+def _current_milliseconds():
+    return int(round(time.time() * 1000))
+
+def wait_for_condition(fn, condition, exceptions_to_ignore=[], codes_to_ignore=[],sleep_time=SLEEP_TIME, timeout=INFINITY) -> bool:
```

Comment:
watch out for using mutable objects as default arguments! It's counter intuitive, but it will use the same instance each time.

```python

def add(to_add, my_list=[]):
    my_list.append(to_add)
    return my_list


>>> print(add(1))
[1]
>>> print(add(2))
[1, 2]
>>> print(add(3))
[1, 2, 3]
>>> print(add(4))
[1, 2, 3, 4]
>>> print(add(5))
[1, 2, 3, 4, 5]
```

for collections and other mutable objects, we should do the following

```python

def add(to_add, my_list=None):
    if my_list is None:
       my_list = []
    my_list.append(to_add)
    return my_list
```

This ensures a new instance scoped to the function each time

```python
>>> print(add(1))
[1]
>>> print(add(2))
[2]
>>> print(add(3))
[3]
>>> print(add(4))
[4]
>>> print(add(5))
[5]
```

Reply:
Thanks I was not aware of that!